### PR TITLE
[Work in progress] Update to PHP 7.1

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | ?
| Fixed issues  | #11397

### CODE REVIEW

File: framework/base/ErrorHandler.php [hunk 1]
```diff
@@ -116,11 +116,8 @@ public function handleException($exception)
                 }
                 exit(1);
             }
-        } catch (\Exception $e) {
-            // an other exception could be thrown while displaying the exception
-            $this->handleFallbackExceptionMessage($e, $exception);
         } catch (\Throwable $e) {
-            // additional check for \Throwable introduced in PHP 7
+            // additional check for \Throwable
```
Comment (Faryshta) [inline framework/base/ErrorHandler.php]: Since the minium requirement is 7.1 how about

`catch(Exception|Throwable $e)`

Comment (cebe) [inline framework/base/ErrorHandler.php]: keep the comment from above instead:

```
// an other exception could be thrown while displaying the exception
```

there is no additional check now.

**Review by cebe — CHANGES_REQUESTED** [review]
requirement in composer.json should be raised.

File: framework/base/ErrorHandler.php (hunk 1 above)
Comment (samdark) [inline framework/base/ErrorHandler.php]: Fixed.

File: composer.json [hunk 2]
```diff
@@ -99,14 +99,14 @@
...
-        "platform": {"php": "5.5"}
+        "platform": {"php": "7.1"}
     },
     "bin": [
         "framework/yii"
     ],
     "extra": {
         "branch-alias": {
-            "dev-master": "2.0.x-dev"
+            "dev-master": "2.1.x-dev"
```
Comment (rob006) [inline composer.json]: Why this was changed?

Comment (samdark) [inline composer.json]: It wasn't https://github.com/yiisoft/yii2/blob/2.1/framework/composer.json#L85

Comment (samdark) [inline composer.json]: Accidentally.

File: .travis.yml [hunk 3]
```diff
@@ -138,12 +96,8 @@ before_script:
   - mysql -e "GRANT ALL PRIVILEGES ON *.* TO 'travis'@'localhost' WITH GRANT OPTION;";
   - psql -U postgres -c 'CREATE DATABASE yiitest;';
 
-  # enable code coverage on PHP 7.1, only one PHP version needs to generate coverage data
-  - |
-    if [ $TRAVIS_PHP_VERSION = '7.1' ]; then
-      PHPUNIT_FLAGS="--coverage-clover=coverage.clover"
-    fi
```
Comment (cebe) [inline .travis.yml]: this should be kept as it now applies to 7.1 and nightly and as soon as 7.2 is out we need it again.

Comment (samdark) [inline .travis.yml]: https://github.com/yiisoft/yii2/commit/f41399a6dc4cf395d6cd0c0d07a98ff93106b240

**Pull request merged**